preview/MsixCore/msixmgr/msixmgr.cpp

```diff
@@ -137,6 +137,167 @@ int main(int argc, char * argv[])
+                        RETURN_IF_FAILED(packages->GetCurrent(&currentPackage));
+                        ComPtr<IAppxManifestPackageId> manifestPackageID;
+                        RETURN_IF_FAILED(currentPackage->GetPackageId(&manifestPackageID));
+
+                        PWSTR packageFullName = nullptr;
+                        RETURN_IF_FAILED(manifestPackageID->GetPackageFullName(&packageFullName));
+
+                        PWSTR packageFileName = nullptr;
+                        RETURN_IF_FAILED(currentPackage->GetFileName(&packageFileName));
+
+                        std::wstring stdPackageFileName = packageFileName;
+                        std::wstring stdPackageFullName = packageFullName;
+
+                        packagesMap[stdPackageFileName] = stdPackageFullName;
+                    }
+
+                    // Use GetPayloadPackages to enumerate over the package files that actually got unpacked
+                    // after applicability logic was applied. 
+                    hasCurrent = false;
+                    ComPtr<IAppxFilesEnumerator> packageFilesEnumerator;
+                    RETURN_IF_FAILED(reader->GetPayloadPackages(&packageFilesEnumerator));
+                    RETURN_IF_FAILED(packageFilesEnumerator->GetHasCurrent(&hasCurrent));
+                    while (hasCurrent)
+                    {
+                        ComPtr<IAppxFile> file;
+                        RETURN_IF_FAILED(packageFilesEnumerator->GetCurrent(&file));
+
+                        PWSTR packageFileName = nullptr;
+                        RETURN_IF_FAILED(file->GetName(&packageFileName));
+
+                        std::wstring stdPackageFileName = packageFileName;
+                        auto packageFullName = packagesMap[packageFileName];
+                        std::wstring packageFolderName = unpackDestination + L"\\" + packageFullName;
+                        packageFolders.push_back(packageFolderName);
+
+                        RETURN_IF_FAILED(packageFilesEnumerator->MoveNext(&hasCurrent));
+                    }
+                }
+
+            }
+            else
+            {
+                return E_INVALIDARG;
+            }
+
+            HMODULE applyACLsDll;
+            applyACLsDll = (LoadLibrary(L"applyacls.dll"));
+            if (applyACLsDll == nullptr)
+            {
+                std::wcout << "Failed to load applyacls.dll. Please confirm the dll is next to this exe" << std::endl;
+            }
+
+            typedef HRESULT(STDMETHODCALLTYPE *APPLYACLSTOPACKAGEFOLDER)(PCWSTR folderPath);
+
+            APPLYACLSTOPACKAGEFOLDER ApplyACLsToPackageFolder =
+                reinterpret_cast<APPLYACLSTOPACKAGEFOLDER>
+                (GetProcAddress(applyACLsDll, "ApplyACLsToPackageFolder"));
+
+            // Update this to pass folder names to API that apply ACLs to those folders
+            for (auto folder : packageFolders)
+            {
+                std::wcout << folder << std::endl;

```

Comment:
>std::wcout << folder << std::endl; [](start = 16, length = 34)

is this debug output or intended output for user's consumption?

Reply:
This is debug output - should we be logging this to the trace logging provider instead? This could potentially be a lot of output.

Reply:
yes, log it to the trace logging provider instead

---
In reply to: [314419946](https://github.com/microsoft/msix-packaging/pull/201#discussion_r314419946) [](ancestors = 314419946)